state/upgrades.go

```diff
@@ -2836,3 +2836,21 @@ func RemoveControllerConfigMaxLogAgeAndSize(pool *StatePool) error {
 	}
 	return nil
 }
+
+// IncrementTasksSequence adds 1 to the "tasks" sequence.
+// Previously, numbering started at 0, now it starts at 1
+// so we need to ensure that upgraded controllers do not
+// get a conflicting task id.
+func IncrementTasksSequence(pool *StatePool) error {
```

Comment:
From memory actions IDs started at 1. Now I didn't like that but is this inconsistent? or do other things start at 0 and actions was the odd one out.

Reply:
they started at 1 because we used sequence returning 0 for the first call and then we added 1 outside that call. Now we use sequenceWithMin and that returns a number starting at 1. So we don't add 1 anymore so that can lead to a conflict the first time after upgrade.